azavea/tilegarden
# Feature/local dev environment

## Overview

This PR sets up a local development environment for the project with the following features:

- Yarn to manage packages and scripts,
- Babel to transpile any ES6 code,
- ESLint for linting,
- Nodemon to hot-reload code as it is written,
- Claudia as a deployment utility,
- Claudia Local API for local lambda testing,
- Runs in a Docker container along with containers that handle a PostGIS data store and an nginx proxy.

## Notes
Serving image files from a local directory is currently not functional due to HTTP header issues. 

## Testing Instructions

 * After a clone, run `/scripts/update` to populate the database and install node modules.
 * Run `./scripts/server` to run the development server, which will be accessible on port **9001**, not 3000 as listed by the output from claudia-local-api. Several example endpoints have been defined in `src/api.js`.
 * To publish, make sure you have specified valid AWS credentials and have listed the proper database credentials in a `.env` file in the root of the project. You can then run `./scripts/publish` to publish to AWS Lambda.


Resolves #8
